net/net.go

```diff
@@ -293,6 +293,27 @@ func (n *net) AddThread(ctx context.Context, addr ma.Multiaddr, opts ...core.New
 		return
 	}
 
+	threadComp, err := ma.NewComponent(thread.Name, id.String())
+	if err != nil {
+		return
+	}
+	peerAddr := addr.Decapsulate(threadComp)
+	addri, err := peer.AddrInfoFromP2pAddr(peerAddr)
+	if err != nil {
+		return
+	}
+
+	// Check if we're trying to dial ourselves (regardless of addr)
+	addFromSelf := addri.ID.Pretty() == n.host.ID().Pretty()
+	if addFromSelf {
+		// Error if we don't have the thread locally
+		if _, err = n.store.GetThread(id); errors.Is(err, lstore.ErrThreadNotFound) {
```

Comment:
If we are dialing ourselves, but we don't have the thread, that's no good. Throw error.